### ISSUE TITLE
Refactor useNostr hook

### ISSUE THREAD

**Issue by ugur-eren** [keep-starknet-strange/joyboy#99]
### OD Hack

This issue is part of ODHack 4.0:

description - https://x.com/OnlyDust_com/status/1791035111992934563
guidelines - https://onlydust.notion.site/ODHack-Common-Guidelines-Project-leads-5e21836f85ad493f9ca429770af0ad40

Please read the guidelines and make sure you follow them.

Please write your thoughts and how long you think it will take to complete the issue in the comments.

### Refactor useNostr hook

[useNostr hook](https://github.com/keep-starknet-strange/joyboy/blob/main/JoyboyCommunity/src/hooks/useNostr.tsx) should have it's own Context and Provider using React Context API. This will allow us to use it in multiple components without having to create multiple instances of the pool and ndk.
Hook should use [Tanstack Query](https://tanstack.com/query/latest) to fetch data from the Nostr pool and mutate the pool. This means it is required to setup Tanstack Query, create a new Context for the pool and ndk, and refactor **every** usage of the hook.

In short, these tasks must be done:
- Setup Tanstack Query
- Create a new Context for the pool and ndk
- Create new hooks to fetch and mutate data by following the useNostr hook
- Refactor every usage of the useNostr hook to use new hooks

Please battle test it before doing a PR.

### Communication

https://t.me/JoyboyStarknet/949

May the Joy be with you!

**Comment by vibenedict** [comment]
Hi, can i work on this  issue

**Comment by ugur-eren** [comment]
Hi @Ayoazeez26 , @vibenedict how much time would it take for you to complete this issue?

**Comment by Ayoazeez26** [comment]
It should take me an estimate of 3 working days to get this complete @ugur-eren

**Comment by vibenedict** [comment]
I will be done in 72hrs
